fix a bug in reverse ReplicaLookup

tbg opened the pull request:
the code assumed that the extra descriptor was directly
adjacent to ours, but intents can cause a gap.

tamird commented:
Can you add a test for this?

spencerkimball commented on storage/replica_command.go:
```diff
@@ -623,8 +623,9 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 	if args.Reverse {
-		// Checks rds[0]. In case 2 we need to remove first range descriptor.
-		if args.Key.Equal(keys.RangeMetaKey(rds[0].StartKey)) {
+		// We had to carry out a single ascending scan above, and in case two
```
What does "case two" refer to?

tschottdorf replied:
I've reworded and removed the reference to the case which belongs to the reverse scan logic further up in the method (outside the diff).

thundercw commented on storage/replica_command.go:
```diff
@@ -623,8 +623,9 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
-		if args.Key.Equal(keys.RangeMetaKey(rds[0].StartKey)) {
+		// We had to carry out a single ascending scan above, and in case two
+		// we might've ended up with descriptor we need to discard now.
+		if !rds[0].StartKey.Less(args.Key) {
```
The `rds[0].StartKey` needs to be encoded as RangeMetaKey.

tschottdorf replied:
done.

tbg committed:
fix a bug in reverse ReplicaLookup

the code assumed that the extra descriptor was directly
adjacent to ours, but intents can cause a gap.

tschottdorf commented:
@tamird I'll look into it.

tschottdorf commented:
going to merge this for CI happiness but test is coming up soon enough.
